### DISCUSSION TITLE
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe8 in position 63: invalid continuation byte

### DISCUSSION THREAD

**Discussion by dounia43** [UMEP-dev/UMEP#374] [Q&A]
I have the following error message every time I try to open a meteorological data file (loaded from the Copernicus program) using the plugin UMEP in QGIS. How can I fix it? I am a beginner in Python.

the meteo file is attached here,
[meteo.zip](https://github.com/UMEP-dev/UMEP/files/8078569/meteo.zip)

UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe8 in position 63: invalid continuation byte 
Traceback (most recent call last):
  File "C:/Users/pc/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP\MetdataProcessor\metdata_processor.py", line 213, in import_file
    header = f.readline().split(delim)
  File "C:\PROGRA~1\QGIS3~1.16\apps\Python37\lib\codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe8 in position 63: invalid continuation byte

**Comment by biglimp** [comment]
Please attach your file so that I could tryit out.

**Comment by dounia43** [comment]
the meteo file is attached here,
[meteo.zip](https://github.com/UMEP-dev/UMEP/files/8078922/meteo.zip)

**Comment by biglimp** [comment]
You are trying to use a netCDF-file directly into UMEP. That is not possible. You should convert it into a text file and then run it through a MetDataProcessor OR download ERA5 data via the ERA5 Download tool in UMEP which will convert nc-files directly into UMEP-formatted meteorological files.

**Reply by dounia43** [reply]
yes sir, i used the ERA5 Download tool to download my data, but the result is a nc-file not converted into UMEP-formatted files.

this is the request
{
  "area": [
    40.375,
    -3.75,
    40.375,
    -3.75
  ],
  "day": [
    "01",
    "02",
    "03",
    "04",
    "05",
    "06",
    "07",
    "08",
    "09",
    "10",
    "11",
    "12",
    "13",
    "14",
    "15",
    "16",
    "17",
    "18",
    "19",
    "20",
    "21",
    "22",
    "23",
    "24",
    "25",
    "26",
    "27",
    "28",
    "29",
    "30",
    "31"
  ],
  "format": "netcdf",
  "grid": [
    0.125,
    0.125
  ],
  "month": [
    "01"
  ],
  "product_type": "reanalysis",
  "time": [
    "00:00",
    "01:00",
    "02:00",
    "03:00",
    "04:00",
    "05:00",
    "06:00",
    "07:00",
    "08:00",
    "09:00",
    "10:00",
    "11:00",
    "12:00",
    "13:00",
    "14:00",
    "15:00",
    "16:00",
    "17:00",
    "18:00",
    "19:00",
    "20:00",
    "21:00",
    "22:00",
    "23:00"
  ],
  "variable": [
    "10m_u_component_of_wind",
    "10m_v_component_of_wind",
    "2m_dewpoint_temperature",
    "2m_temperature",
    "geopotential",
    "surface_pressure",
    "surface_solar_radiation_downwards",
    "surface_thermal_radiation_downwards",
    "surface_sensible_heat_flux",
    "surface_latent_heat_flux",
    "surface_net_solar_radiation",
    "surface_net_thermal_radiation",
    "total_precipitation",
    "forecast_albedo",
    "forecast_surface_roughness",
    "friction_velocity"
  ],
  "year": [
    "2010"
  ]
}

**Reply by biglimp** [reply]
The tool should have created the nc-files AND a text-file. Are you sure that no text file was created? Can you make a screen dump of the settings when using ERA5 Download.

https://umep-docs.readthedocs.io/en/latest/pre-processor/Meteorological%20Data%20Download%20data%20(ERA5).html

**Reply by dounia43** [reply]
you are right sir, I found my error. thank you so much.
